Bootstraping app with Beacon integration

SiddharthV1 opened the pull request:
resolves #124 

https://user-images.githubusercontent.com/424818/200527094-a02e1be1-9b01-4194-8f8b-21c8b1d32148.mp4

Commit:
Default react app using create-react-app

Commit:
Added beacon dependency

Commit:
Nextjs Project structure plus beacon support

Commit:
Broken stock react app, webpack issue with beacon

Commit:
Resolved webpack issue

Commit:
WIP context to store wallet

Commit:
Deleted nextjs app

Commit:
Added a wallet context which will hold the client

Commit:
Hook for wallet context

Commit:
Code to refresh wallet connection status

Commit:
Selective renderers based on wallet's status

Commit:
Renamed wallet util file to beacon

Commit:
WIP minimal ui

Commit:
WIP code for unstyled ui

Commit:
Setter for dApp client passed to wallet's context

Commit:
WIP Loading beacon client in context

Commit:
Wallet disconnection function added to context

Commit:
WIP loading wallet address in context

Commit:
Wallet address stored in context

Commit:
Disconnect wallet feature added

Commit:
...

Commit:
Updated readme

vishakh commented on V2/tezex/config-overrides.js:
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable no-param-reassign */
+// eslint-disable-next-line import/no-extraneous-dependencies
+const webpack = require('webpack');
```
webpack is not mentioned in package.json.

SiddharthV1 replied:
> tezex@0.1.0 build
> react-scripts build


There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "webpack": "4.44.2"

Don't try to install it manually: your package manager does it automatically.
However, a different version of webpack was detected higher up in the tree:

  /home/siddharth/nix-dev-1/tezex/TEZEX/V2/tezex/node_modules/webpack (version: 5.74.0)

Manually installing incompatible versions is known to cause hard-to-debug issues.

If you would prefer to ignore this check, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That will permanently disable this message but you might encounter other issues.

To fix the dependency tree, try following the steps below in the exact order:

  1. Delete package-lock.json (not package.json!) and/or yarn.lock in your project folder.
  2. Delete node_modules in your project folder.
  3. Remove "webpack" from dependencies and/or devDependencies in the package.json file in your project folder.
  4. Run npm install or yarn, depending on the package manager you use.

In most cases, this should be enough to fix the problem.
If this has not helped, there are a few other things you can try:

  5. If you used npm, install yarn (http://yarnpkg.com/) and repeat the above steps with it instead.
     This may help because npm has known issues with package hoisting which may get resolved in future versions.

  6. Check if /home/siddharth/nix-dev-1/tezex/TEZEX/V2/tezex/node_modules/webpack is outside your project directory.
     For example, you might have accidentally installed something in your home folder.

  7. Try running npm ls webpack in your project folder.
     This will tell you which other package (apart from the expected react-scripts) installed webpack.

If nothing else helps, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That would permanently disable this preflight check in case you want to proceed anyway.

SiddharthV1 replied:
that's the issue when I install webpack

vishakh commented on V2/tezex/package.json:
```diff
@@ -0,0 +1,45 @@
+{
```
This is what I see when I run `npm i`:

```
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated rollup-plugin-babel@4.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.

added 2041 packages, and audited 2042 packages in 9s

180 packages are looking for funding
  run `npm fund` for details

32 vulnerabilities (1 low, 1 moderate, 20 high, 10 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

Could we clean this up a bit?

vishakh commented on V2/tezex/src/App.tsx:
```diff
@@ -0,0 +1,21 @@
+import React, { FC } from "react";
```
I get a `TS1259` here.

vishakh commented on V2/tezex/src/components/session/WalletConnected.tsx:
```diff
@@ -0,0 +1,12 @@
+import { FC } from "react";
+import { useSession} from "../../hooks/session";
+
+export const WalletConnected: FC = (props) => {
```
In order to properly use TypeScript, could we please properly type `props` everywhere?

This produces a `TS2339` for me.

vishakh commented on V2/tezex/src/components/wallet/Wallet.tsx:
```diff
@@ -0,0 +1,31 @@
+import { FC } from "react";
+import React, { useEffect } from "react";
```
TS1259: Module '"/home/vishakh/dev/cryptonomic/TEZEX/V2/tezex/node_modules/@types/react/index"' can only be default-imported using the 'esModuleInterop' flag  index.d.ts(62, 1): This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

vishakh commented on V2/tezex/src/contexts/wallet.tsx:
```diff
@@ -0,0 +1,12 @@
+import { createContext, useState } from "react";
+
+export interface WalletInfo {
+	client: any,
+	setClient: any,
+	address: any,
+	setAddress: any,
+	disconnect: any
```
Could we please avoid `any` as much as possible and properly type everything?

vishakh commented on V2/tezex/src/hooks/wallet.ts:
```diff
@@ -0,0 +1,9 @@
+import { useContext } from "react";
+import { WalletContext , WalletInfo } from "../contexts/wallet";
```
`WalletInfo` in unused.

vishakh commented on V2/tezex/src/index.tsx:
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
```
TS1259 error here.

vishakh commented on V2/tezex/src/index.tsx:
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
```
TS1192 error here.

Commit:
Importing project config from tezfin

The current project has webpack issues due to conflict with react and
react script versions , thus using the tezfin project setup

Commit:
WIP build issues

Commit:
Imported Tezori package.json + tsconfig

Webpack issue still exists

Commit:
Potentially fixed webpack-issue

Commit:
Merge branch 'beacon-webpack' into beacon

Commit:
Potentially resolved webpack issue

Commit:
Layout and MainWindow components have typed props.

Commit:
Heading Interface

Commit:
delete old config

Commit:
Temp remove props from header

Commit:
Connection Interface

Commit:
Disconnection Interface

Commit:
ISession Interface

Commit:
Removing merge artifacts from package.json

Commit:
Resolved build errors

Commit:
Fixed build warnings

Commit:
Fixed wallet rendering issue when aborting connect

Commit:
Type for session context

Commit:
Remove artifact
